### PR TITLE
Added recipe for Daggy - Data Aggregation Utility and C/C++ developer library for data streams catching

### PR DESCRIPTION
**Pull request by synacker**
daggy/2.1.2

I'm the author of Daggy - Data Aggregation Utility and C/C++ developer library for crossplatform and serverless streams catching.

In release 2.1 the conan package for library were added.

---

- [ x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by synacker** [comment]
@prince-chrismc Well, I did it ) I built package on all platforms and this was a nightmare. Do you have any suggestions for this pull request?)

File: recipes/daggy/2.1/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        "with_console": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "with_ssh2": True,
+        "with_yaml": True,
+        "with_console": False,
+        "shared": False,
+        "fPIC": False
```
Comment (uilianries) [inline recipes/daggy/2.1/conanfile.py:29]:
```suggestion
        "fPIC": True
```

By default, it should be true.

File: recipes/daggy/2.1/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    def package(self):
+        cmake = self._configure()
+        cmake.install()
+
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.libs = ["DaggyCore"]
+
+    def imports(self):
```
Comment (uilianries) [inline recipes/daggy/2.1/conanfile.py:130]: Why do you need to import something here?

File: recipes/daggy/2.1/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.21.0)
+project(DaggyPackageTest)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (uilianries) [inline recipes/daggy/2.1/test_package/CMakeLists.txt:5]:
```suggestion
conan_basic_setup(TARGETS)
```

So cmake generator won't mess the cmake targets generated by Conan

File: recipes/daggy/2.1/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+find_package(daggy REQUIRED)
+
+if (NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 17)
+endif()
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+add_subdirectory(testcpp)
```
Comment (uilianries) [inline recipes/daggy/2.1/test_package/CMakeLists.txt:14]:
```suggestion
```

Keep all under test_package folder, don't need to create subfolders for a simple test.

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution, please, take a look on my review.

**Review by prince-chrismc — COMMENTED** [review]
I forgot to click submit I am sorry

image

File: recipes/daggy/2.1/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (synacker) [reply recipes/daggy/2.1/conanfile.py]: Yes, I removed it, thank you

File: recipes/daggy/2.1/test_package/CMakeLists.txt (hunk 4 above)
Comment (synacker) [reply recipes/daggy/2.1/test_package/CMakeLists.txt]: Moved test from subdirectory

[the remark this answers is not in the document]
Comment (synacker) [reply recipes/daggy/2.1/test_package/CMakeLists.txt]: I did this

File: recipes/daggy/2.1/conanfile.py (hunk 1 above)
Comment (synacker) [reply recipes/daggy/2.1/conanfile.py]: Got an error - [HOOK - conan-center.py] pre_build(): ERROR: [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not managed correctly. Please remove it for shared option: del self.options.fPIC (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H007)

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**